pkg/patterns/addon/pkg/status/conditions.go

```diff
@@ -9,19 +9,34 @@ import (
 )
 
 const (
-	AbnormalReason = "ContainAbnormalTrueConditions"
+	AbnormalReason = "AbnormalTrueConditions"
 	NormalReason   = "Normal"
 	ReadyType      = "Ready"
 )
 
-// GetCondition returns the first met Condition whose type matches the given condition type.
-func GetCondition(conditions []*metav1.Condition, condType status.ConditionType) *metav1.Condition {
-	for i, condition := range conditions {
-		if condition.Type == string(condType) {
-			return conditions[i]
-		}
+// SetInProgress set the present condition to a single condition with type "Ready" and status "false". This means
+// the current resources is still reconciling. If any deployment manifests are abnormal, their abnormal status condition will
+// be recorded in the `message` field.
+func SetInProgress(commonStatus *addonsv1alpha1.CommonStatus, abnormalTrueConditions []status.Condition) {
+	setCondition(metav1.ConditionFalse, commonStatus, abnormalTrueConditions)
+}
+
+// SetReady set the present condition to a single condition with type "Ready" and status "true". This means
+// all the deployment manifests are reconciled.
+func SetReady(commonStatus *addonsv1alpha1.CommonStatus, abnormalTrueConditions []status.Condition) {
+	setCondition(metav1.ConditionTrue, commonStatus, abnormalTrueConditions)
+}
+
+func setCondition(status metav1.ConditionStatus, commonStatus *addonsv1alpha1.CommonStatus, abnormalTrueConditions []status.Condition) {
+	reason, message := humanMessagefromConditions(abnormalTrueConditions)
+	commonStatus.Conditions = []*metav1.Condition{{
+		Status:             status,
+		Type:               ReadyType,
+		LastTransitionTime: metav1.Now(),
```

Comment:
Does this correctly update the `LastTransitionTime` only when the actual value of the `Status` field changes? It might be useful to use the functions available in `k8s.io/apimachinery/pkg/api/meta` for updating the conditions.

Reply:
Ah, I thought the transition refers to the present reconciliation cycle (where users can use to see when the object was last processed by the controller).  "the actual value of the Status field changes", that makes a lot of sense.